components/gitpod-protocol/src/util/tracing.ts

```diff
@@ -32,28 +32,22 @@ export namespace TraceContext {
         return opentracing.globalTracer().startSpan(operation, options);
     }
 
-    export function childContextWithSpan(operation: string, parentCtx: TraceContext): TraceContextWithSpan {
+    export function childContext(operation: string, parentCtx: TraceContext): TraceContextWithSpan {
         const span = startSpan(operation, parentCtx);
         return { span };
     }
 
-    export function startAsyncSpan(operation: string, ctx: TraceContext): opentracing.Span {
-        const options: opentracing.SpanOptions = {};
-        if (!!ctx.span) {
-            options.references = [opentracing.followsFrom(ctx.span.context())];
-        }
-        return opentracing.globalTracer().startSpan(operation, options);
-    }
-
-    export function logError(ctx: TraceContext, err: Error) {
+    export function setError(ctx: TraceContext, err: Error) {
```

Comment:
For methods that only need access to the span I don't think we need to pass in the full context. But that would mean also changing addNestedTags so out of scope of this PR I think

Reply:
We always pass `TraceContext`, and I feel we should keep the pattern. Especially as it's just a `{ }` wrapper.